### PR TITLE
Update the WordPress packages with Gutenberg 15.1.0 changes

### PR DESCRIPTION
**Pull request by youknowriad**
This PR updates the WordPress packages with the latest released ones from Gutenberg 15.1.0 

While updating the packages, it was necessary to bring the following changes as well:

 - React upgrade to 18.2.0
 - iframe the block theme editor. (Related PRs https://github.com/WordPress/wordpress-develop/pull/3904, https://github.com/WordPress/wordpress-develop/pull/3903 )

Trac ticket: https://core.trac.wordpress.org/ticket/57471

### CODE REVIEW

**Comment by Mamaduka** [comment]
I noticed that we have a few methods with the `gutenberg_` prefix in the `block-library` packages. I'm looking into this.

File: src/wp-includes/blocks/gallery.php
Comment (felixarntz) [inline src/wp-includes/blocks/gallery.php:109]: This can be simplified:
```suggestion
	$gallery_styles = array(
		array(
			'selector'     => ".wp-block-gallery.{$unique_gallery_classname}",
			'declarations' => array(
				'--wp--style--unstable-gallery-gap' => $gap_column,
				'gap'                               => $gap_value,
			),
		),
	);
```

File: src/wp-includes/blocks/calendar.php
Comment (felixarntz) [inline src/wp-includes/blocks/calendar.php]: See @Mamaduka's feedback:
```suggestion
	$styles        = wp_style_engine_get_styles( array( 'color' => $color_block_styles ), array( 'convert_vars_to_classnames' => true ) );
```

File: src/wp-includes/blocks/gallery.php
Comment (felixarntz) [inline src/wp-includes/blocks/gallery.php]: See above. There are a few more of these in the PR that need to be replaced.
```suggestion
	wp_style_engine_get_stylesheet_from_css_rules(
```

**Review by felixarntz — COMMENTED** [review]
@youknowriad Mostly looks good, just one nit-pick and the remaining `gutenberg_` prefixes that need to be replaced. Is that something you can just update in this PR or does it require a fix on the Gutenberg side?

Other than that, I assume the intention is to commit this once #3920 has been committed?

**Comment by Mamaduka** [comment]
@felixarntz, I already fixed `gutenberg_` prefix issue in Gutenberg repo. See https://github.com/WordPress/gutenberg/pull/47649.

@ntsekouras and I will start working on package updates that include recent fixes.

**Comment by youknowriad** [comment]
> Mostly looks good, just one nit-pick and the remaining gutenberg_ prefixes that need to be replaced. Is that something you can just update in this PR or does it require a fix on the Gutenberg side?

Actually, I can't touch these files here, even the nitpicks suggested here because every time someone will do `npm install` it will revert these changes locally causing confusion. So I think we should instead make a PR to Gutenberg and wait for the next round of package release.

**Comment by youknowriad** [comment]
I've just noticed that there's a blocks fixtures tests in Core right now. Anyone know why this test has been moved from Gutenberg to here? It's just going to add maintenance burden for us for a small value since all the code for blocks is coming through package. In other words, these tests are better by their code (in the Gutenberg repo).

**Comment by youknowriad** [comment]
Ok I've managed to fix all the failing tests here. It's not perfect but It should be good to land to unblock the PRs that depend on it.

I've noticed a small style glitch on the empty post placeholder. Something to keep an eye on after all the backports.

**Review by felixarntz — APPROVED** [review]
@youknowriad LGTM. Would still be nice to address https://github.com/WordPress/wordpress-develop/pull/3914/files#r1095179123, though that is not a blocker.

**Comment by youknowriad** [comment]
@felixarntz It needs to be addressed on Gutenberg, that way it comes in the next package update. You can't make changes to these files in Core.

**Comment by felixarntz** [comment]
@youknowriad Ah right, forgot about that, thanks!

**Review by ntsekouras — APPROVED** [review]
LGTM to land this. Thank you Riad!

**Comment by youknowriad** [comment]
This has been committed

**Comment by Mamaduka** [comment]
Thank you, Riad!

The changeset - https://core.trac.wordpress.org/changeset/55246.

**Pull request closed without merging**
